source/common/http/conn_pool_grid.h

```diff
@@ -182,6 +181,9 @@ class ConnectivityGrid : public ConnectionPool::Instance,
   // Set the required idle callback on the pool.
   void setupPool(ConnectionPool::Instance& pool);
 
+  // Return origin if the remote host has IP address.
+  absl::optional<AlternateProtocolsCache::Origin> originFromHost() const;
```

Comment:
Let's make this private. It looks like we use it below in a test, but we can just as easily create an Origin explicitly in that test.

nit: consider just naming this method origin() or getOrigin().